java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java

```diff
@@ -25,40 +26,62 @@ interface UnleashFFI extends Library {
 
     Pointer take_state(Pointer ptr, String toggles);
 
-    boolean check_enabled(Pointer ptr, String name, String context);
+    Pointer check_enabled(Pointer ptr, String name, String context);
 
     Pointer check_variant(Pointer ptr, String name, String context);
 
```

Comment:
We'll need the metrics methods here too but that can be in another PR

Reply:
Yes, will cover that here: https://linear.app/unleash/issue/2-1560/implement-metrics